1. Faça um Programa que peça dois números e imprima o maior deles.

In [2]:
num1 = input('Digite um número')
num2 = input('Digite outro número')

if num1 > num2:
    print(f'{num1} é maior')
else:
    print(f'{num2} é maior')

Digite um número6
Digite outro número1
6 é maior


2. Faça um Programa que verifique se uma letra digitada é vogal ou consoante.

In [6]:
vogal='aeiou'
letra = input('Digite uma letra para saber se é vogal ou consoante')

if letra in vogal:
    print('É uma vogal')
else:
    print('É uma consoante')

Digite uma letra para saber se é vogal ou consoantee
É uma vogal


3. Faça um programa para a leitura de duas notas parciais de um aluno. O programa deve calcular a média alcançada por aluno e apresentar:
- A mensagem "Aprovado", se a média alcançada for maior ou igual a sete;
- A mensagem "Reprovado", se a média for menor do que sete;
- A mensagem "Aprovado com Distinção", se a média for igual a dez.

In [13]:
nota1 = int(input('Digite a primeira nota do aluno'))
nota2 = int(input('Digite a segunda nota do aluno'))

media = (nota1 + nota2)/2
if media < 7:
    print(f'Aluno REPROVADO com média:{media}')
elif media > 6 and media <10:
    print (f'Aluno APROVADO com média:{media}')
elif media == 10:
    print(f'Aluno APROVADO com Distinção, com média:{media}')

Digite a primeira nota do aluno10
Digite a segunda nota do aluno10
Aluno APROVADO com Distinção, com média:10.0


4. Faça um Programa que leia três números e mostre-os em ordem decrescente.

In [3]:
n1 = input('Digite um número')
n2 = input('Digite outro número')
n3 = input('Digite outro número')

if n1 > n2 and n1 > n3:
    if n2 > n3:
        print(n1, n2, n3)
    else:
        print(n1,n3,n2)
elif n2 > n1 and n2 > n3:
    if n1 > n3:
        print(n2, n1, n3)
    else:
        print(n2,n3,n1)
else:
    if n2 > n1:
        print(n3, n2, n1)
    else:
        print(n3,n1,n2)

Digite um número120
Digite outro número236
Digite outro número458
458 236 120


5. As Organizações Tabajara resolveram dar um aumento de salário aos seus colaboradores e lhe contrataram para desenvolver o programa que calculará os reajustes.
Faça um programa que recebe o salário de um colaborador e o reajuste segundo o seguinte critério, baseado no salário atual:
- salários até R$ 280,00 (incluindo) : aumento de 20%
- salários entre R$ 280,00 e R$ 700,00 : aumento de 15%
- salários entre R$ 700,00 e R$ 1500,00 : aumento de 10%
- salários de R$ 1500,00 em diante : aumento de 5% Após o aumento ser realizado,
informe na tela:
- o salário antes do reajuste;
- o percentual de aumento aplicado;
- o valor do aumento;
- o novo salário, após o aumento.

In [2]:
salario = int(input('Digite o salário de uma colaborador '))

def calculaTudo(percentual, ajuste):
    novo_salario = salario + ajuste
    print(f'Salario antes do reajuste: R${salario}\nPercentual é:{percentual}\nValor do aumento: R${ajuste}\nNovo salário: R${novo_salario}')

if salario <= 280:
    percentual = '20%'
    ajuste = salario * 0.2
    calculaTudo(percentual, ajuste)
elif salario > 280 and salario <=700:
    percentual = '15%'
    ajuste = salario * 0.15
    calculaTudo(percentual, ajuste)
elif salario > 780 and salario < 1500:
    percentual = '10%'
    ajuste = salario * 0.1
    calculaTudo(percentual, ajuste)
else:
    percentual = '5%'
    ajuste = salario * 0.05
    calculaTudo(percentual, ajuste)

Digite o salário de uma colaborador 1000
Salario antes do reajuste: R$1000
Percentual é:10%
Valor do aumento: R$100.0
Novo salário: R$1100.0


6. Faça um Programa que leia um número e exiba o dia correspondente da semana.
(1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.

In [8]:
dia = int(input('Digite um número para saber o dia da semana correspondente: '))

if dia == 1:
    print('Domingo')
elif dia == 2:
    print('Segunda-feira')
elif dia == 3:
    print('Terça-feira')
elif dia == 4:
    print('Quarta-feira')
elif dia == 5:
    print('Quinta-feira')
elif dia == 6:
    print('Sexta-feira')
elif dia == 7:
    print('Sábado')
else:
    print('Valor digitado é inválido')

Digite um número para saber o dia da semana correspondente: 10
Valor digitado é inválido


7. Faça um programa que lê as duas notas parciais obtidas por um aluno numa
disciplina ao longo de um semestre, e calcule a sua média. A atribuição de conceitos
obedece à tabela abaixo:
Média de Aproveitamento Conceito
 Entre 9.0 e 10.0 A
 Entre 7.5 e 9.0 B
 Entre 6.0 e 7.5 C
 Entre 4.0 e 6.0 D
 Entre 4.0 e zero E
 O algoritmo deve mostrar na tela as notas, a média, o conceito correspondente e a
mensagem “APROVADO” se o conceito for A, B ou C ou “REPROVADO” se o conceito
for D ou E.

In [9]:
nota1 = int(input('Digite a primeira nota do aluno(a)'))
nota2 = int(input('Digite a segunda nota do aluno(a)'))

media = (nota1 + nota2)/2

if media >= 9 and media <= 10:
    print(f' Nota 1: {nota1}\n Nota 2: {nota2}\n Média: {media}\n Conceito: A-Aprovado')
elif media >= 7.5 and media <9:
    print(f' Nota 1: {nota1}\n Nota 2: {nota2}\n Média: {media}\n Conceito: B-Aprovado')
elif media >= 6 and media <7.5:
    print(f' Nota 1: {nota1}\n Nota 2: {nota2}\n Média: {media}\n Conceito: C-Aprovado')
elif media >= 4 and media <6:
    print(f' Nota 1: {nota1}\n Nota 2: {nota2}\n Média: {media}\n Conceito: D-Reprovado')
elif media >= 0 and media <4:
    print(f' Nota 1: {nota1}\n Nota 2: {nota2}\n Média: {media}\n Conceito: E-Reprovado')

Digite a primeira nota do aluno(a)3
Digite a segunda nota do aluno(a)3
 Nota 1: 3
 Nota 2: 3
 Média: 3.0
 Conceito: E-Reprovado


8. Faça um Programa que peça os 3 lados de um triângulo. O programa deverá informar se os valores podem ser um triângulo. Indique, caso os lados formem um triângulo, se o mesmo é: equilátero, isósceles ou escaleno.
Dicas:
- Três lados formam um triângulo quando a soma de quaisquer dois lados for maior que o terceiro;
- Triângulo Equilátero: três lados iguais;
- Triângulo Isósceles: quaisquer dois lados iguais;
- Triângulo Escaleno: três lados diferentes;

In [21]:
lado1 = int(input('Digite o primeiro lado do triângulo'))
lado2 = int(input('Digite o segundo lado do triângulo'))
lado3 = int(input('Digite o terceiro lado do triângulo'))

if lado1 > (lado2 + lado3) or lado2 > (lado3 + lado1) or lado3 > (lado1 + lado2):
    print('Não é um triângulo')
else:
    if lado1 == lado2 == lado3:
        print('Triângulo Equilátero')
    elif lado1 == lado2 or lado2 == lado3 or lado3 == lado1:
        print('Triângulo Isóceles')
    else:
        print('triângulo Escaleno')
    



Digite o primeiro lado do triângulo9
Digite o segundo lado do triângulo8
Digite o terceiro lado do triângulo5
triângulo Escaleno


9. Faça um programa que calcule as raízes de uma equação do segundo grau, na forma ax2 + bx + c. O programa deverá pedir os valores de a, b e c e fazer as consistências, informando ao usuário nas seguintes situações:
- Se o usuário informar o valor de A igual a zero, a equação não é do segundo grau e o programa não deve fazer pedir os demais valores, sendo encerrado; Ok
- Se o delta calculado for negativo, a equação não possui raizes reais. Informe ao usuário e encerre o programa; OK
- Se o delta calculado for igual a zero a equação possui apenas uma raiz real;
informe-a ao usuário;
- Se o delta for positivo, a equação possui duas raiz reais; informe-as ao usuário;

In [9]:
import math
a = int(input('Digite o valor de A: '))
b = int(input('Digite o valor de B: '))
c = int(input('Digite o valor de C: '))

if a == 0:
    print('A equação não é do segundo grau')
else:
    delta = math.pow(b,2) - (4*a*c)
    if delta < 0:
        print(f'A equação não possui raízes reais. Delta:{delta}')
    elif delta == 0:
        raiz = ((-1)*b + math.sqrt(delta))/(2*a)
        print(f'Equação {a}x² + {b}x + {c}\nDelta: {delta}\nRaiz da equação: {raiz}')
    elif delta > 0:
        raiz1 = ((-1)*b + math.sqrt(delta))/(2*a)
        raiz2 = ((-1)*b - math.sqrt(delta))/(2*a)
        print(f'Equação {a}x² + {b}x + {c}\nDelta: {delta}\nPrimeira raiz da equação: {raiz1}\nSegunda raiz da equação: {raiz2}')
        
        
    

Digite o valor de A: 1
Digite o valor de B: -5
Digite o valor de C: 6
Equação 1x² + -5x + 6
Delta: 1.0
Primeira raiz da equação: 3.0
Segunda raiz da equação: 2.0


10. Faça um Programa para um caixa eletrônico. O programa deverá perguntar ao usuário a valor do saque e depois informar quantas notas de cada valor serão fornecidas. As notas disponíveis serão as de 1, 5, 10, 50 e 100 reais. O valor mínimo é de 10 reais e o máximo de 600 reais. O programa não deve se preocupar com a quantidade de notas existentes na máquina.
- Exemplo 1: Para sacar a quantia de 256 reais, o programa fornece duas notas de 100, uma nota de 50, uma nota de 5 e uma nota de 1;
- Exemplo 2: Para sacar a quantia de 399 reais, o programa fornece três notas de 100, uma nota de 50, quatro notas de 10, uma nota de 5 e quatro notas de 1.

In [24]:
valor = int(input('Digite o valor em que deseja sacar: '))

if valor >= 10 and valor <=600:
    notas100= valor // 100 
    valor = valor - (notas100 * 100)
    print(f'Quantidade de notas de 100: {notas100}')
    
    notas50 = valor // 50
    valor = valor - (notas50 * 50)
    print(f'Quantidade de notas de 50: {notas50}')
    
    notas10 = valor // 10
    valor = valor - (notas10 * 10)
    print(f'Quantidade de notas de 10: {notas10}')
    
    notas5  = valor // 5
    valor = valor - (notas5 * 5)
    print(f'Quantidade de notas de 5: {notas5}')
    
    notas1  = valor // 1
    print(f'Quantidade de notas de 1: {notas1}')

else:
    print('Não é possível sacar esse valor')

Digite o valor em que deseja sacar: 277
Quantidade de notas de 100: 2
Quantidade de notas de 50: 1
Quantidade de notas de 10: 2
Quantidade de notas de 5: 1
Quantidade de notas de 1: 2


11- Faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As perguntas são:
"Telefonou para a vítima?"
"Esteve no local do crime?"
"Mora perto da vítima?"
"Devia para a vítima?"
"Já trabalhou com a vítima?"
O programa deve no final emitir uma classificação sobre a participação da pessoa no crime. Se a pessoa responder positivamente a 2 questões ela deve ser classificada como "Suspeita", entre 3 e 4 como "Cúmplice" e 5 como "Assassino". Caso contrário, ele será classificado como "Inocente".

In [12]:
perguntas = [ 
    "Telefonou para a vítima?" ,
    "Esteve no local do crime?" ,
    "Mora perto da vítima?" ,
    "Devia para a vítima?" ,
    "Já trabalhou com a vítima?" 
]

resposta = 0
for resp in perguntas:
      resposta += (input(resp) == "s")
    
if resposta == 2:
    print("Suspeito")
elif resposta == 3 or resposta == 4:
    print("Cúmplice")
elif resposta ==5:
    print("Assassino")
else:
    print("Inocente")

Telefonou para a vítima?n
Esteve no local do crime?n
Mora perto da vítima?n
Devia para a vítima?n
Já trabalhou com a vítima?n
Inocente


12. Uma fruteira está vendendo frutas com a seguinte tabela de preços:
Até 5 Kg Acima de 5 Kg
Morango R$ 2,50 por Kg R$ 2,20 por Kg
Maçã R$ 1,80 por Kg R$ 1,50 por Kg
Se o cliente comprar mais de 8 Kg em frutas ou o valor total da compra ultrapassar R$ 25,00, receberá ainda um desconto de 10% sobre este total. Escreva um algoritmo para ler a quantidade (em Kg) de morangos e a quantidade (em Kg) de maças adquiridas e escreva o valor a ser pago pelo cliente.

In [9]:
morango = float(input("Quantos kg de morango ?"))
macas = float(input("Quantos kg de maçãs ?"))

if morango <=5 and macas <= 5:
    preco_morango = 2.50 * morango
    preco_maca = 1.80 * macas
else:
    preco_morango = 2.20 * morango
    preco_maca = 1.50 * macas
    

if (morango + macas) > 8 or(preco_morango + preco_maca) > 25:
     valor_total  = (preco_morango + preco_maca) * 0.9
else:
    valor_total = preco_maca + preco_morango
    
print('Valor total da compra é R$ %.2f' % (valor_total))
    
    

Quantos kg de morango ?5
Quantos kg de maçãs ?8
Valor total da compra é R$ 20.70


13. Faça um programa que peça uma nota, entre zero e dez. Mostre uma mensagem caso o valor seja inválido e continue pedindo até que o usuário informe um valor válido.

In [11]:
while True:
    
    nota = int(input('Digite uma nota entre 0 e 10'))
    if nota >= 0 and nota <= 10:
        break
    else:
        print('Nota inválida')

print(f'Valor válido {nota}')

Digite uma nota entre 0 e 1025
Nota inválida
Digite uma nota entre 0 e 1036
Nota inválida
Digite uma nota entre 0 e 1036
Nota inválida
Digite uma nota entre 0 e 109
Valor válido 9


13. Faça um programa que leia e valide as seguintes informações:
Nome: maior que 3 caracteres;
Idade: entre 0 e 150;
Salário: maior que zero;
Sexo: 'f' ou 'm';
Estado Civil: 's', 'c', 'v', 'd'

In [2]:
while True:
    nome = input('Digite seu nome: ')
    idade = int(input('Digite sua idade: '))
    salario = int(input('Digite seu salario: '))
    sexo = input('Digite seu sexo: f ou m')
    ecivil = input('Estado cívil:\ns - solteiro\nc - casado\nv - víuvo\nd - divorciado\n')
    
    if len(nome) > 3:
        if idade >0 and idade <=150:
            if salario >0:
                if sexo == 'm' or sexo == 'f':
                    if ecivil == 's' or ecivil == 'c' or ecivil == 'v' or ecivil == 'd':
                        print(f'Informações validadas\nNome:{nome}\nIdade:{idade}\nSalário:{salario}\nSexo:{sexo}\nEstado cívil:{ecivil}')
                        break
                    else:
                        print('Estado cívil inválido\n Digite:\ns - solteiro\n c - casado\n v - víuvo\n d - divorciado')
                else:
                    print('Sexo inválido')
            else:
                print('Salário inválido')
        else:
            print('Idade inválida')
    else:
        print('Nome inválido')
                        

Digite seu nome: maria
Digite sua idade: 18000
Digite seu salario: 1200
Digite seu sexo: f ou mf
Estado cívil:
s - solteiro
c - casado
v - víuvo
d - divorciado
d
Idade inválida
Digite seu nome: marina
Digite sua idade: 89
Digite seu salario: 1200
Digite seu sexo: f ou mf
Estado cívil:
s - solteiro
c - casado
v - víuvo
d - divorciado
c
Informações validadas
Nome:marina
Idade:89
Salário:1200
Sexo:f
Estado cívil:c


14. Faça um programa que leia 5 números e informe o maior número

In [6]:
maior  =  int(input("Digite o 1º número:"))
for num in range(2,6):
    x = int(input(f"Digite o {num}º número: "))
    if x > maior:
        maior = x
print(f"O maior número é:{maior}")

Digite o 1º número:567
Digite o 2º número: 1
Digite o 3º número: 56
Digite o 4º número: 78
Digite o 5º número: 12
O maior número é:567


15. Faça um programa que leia 5 números e informe a soma e a média dos números.

In [3]:

soma = 0
for num in range(1,6):
    n = int(input(f"Digite o {num}º número: "))
    soma += n
print(f"A soma dos números é:{soma}")
print(f'A média é:{soma/5}')

Digite o 1º número: 4
Digite o 2º número: 5
Digite o 3º número: 1
Digite o 4º número: 1
Digite o 5º número: 1
A soma dos números é:12
A média é:2.4


16. Faça um programa que imprima na tela apenas os números ímpares entre 1 e 50

In [13]:

for n in range(1,51) :
    if n % 2 !=0:
        print(n)

    

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49


17. Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer número inteiro entre 1 a 10. O usuário deve informar de qual numero ele deseja ver a tabuada. A saída deve ser conforme o exemplo abaixo:
Tabuada de 5:
5 X 1 = 5
5 X 2 = 10
...
5 X 10 = 50

In [16]:
numero = int(input('Digite o número em que deseja ver a tabuada'))
for num in range(1,11):
    print(f'{numero} x {num} = {numero * num}')
    

Digite o número em que deseja ver a tabuada80
80 x 1 = 80
80 x 2 = 160
80 x 3 = 240
80 x 4 = 320
80 x 5 = 400
80 x 6 = 480
80 x 7 = 560
80 x 8 = 640
80 x 9 = 720
80 x 10 = 800


18. A série de Fibonacci é formada pela seqüência 1,1,2,3,5,8,13,21,34,55,... Faça um programa capaz de gerar a série até o n−ésimo termo

In [20]:
a = 0
b = 1
n = int(input("Digite até qual termo deseja ir? "))
for i in range(n):
    print(a)
    aux = b
    b = a + b
    a = aux

Digite até qual termo deseja ir? 8
0
1
1
2
3
5
8
13


19. Faça um programa que calcule o fatorial de um número inteiro fornecido pelo usuário. Ex.: 5!=5.4.3.2.1=120

In [23]:
fatorial = int(input('Digite um número para saber o seu fatorial: '))
for f in range(1, fatorial):
    
    fatorial = fatorial * f
    
print(f'É {fatorial}')

Digite um número para saber o seu fatorial: 5
É 120


20. O Sr. Manoel Joaquim possui uma grande loja de artigos de R$ 1,99, com cerca de 10 caixas. Para agilizar o cálculo de quanto cada cliente deve pagar ele desenvolveu um tabela que contém o número de itens que o cliente comprou e ao lado o valor da conta. Desta forma a atendente do caixa precisa apenas contar quantos itens o cliente está levando e olhar na tabela de preços. Você foi contratado para desenvolver o programa que monta esta tabela de preços, que conterá os preços de 1 até 50 produtos, conforme o exemplo abaixo:
Lojas Quase Dois - Tabela de preços
1 - R$ 1.99
2 - R$ 3.98
...
50 - R$ 99.50


In [24]:
for i in range(1, 51):
    preco = i * 1.99
    print(f"{num} - R$ {preco}")

1 - R$ 1.99
2 - R$ 3.98
3 - R$ 5.97
4 - R$ 7.96
5 - R$ 9.95
6 - R$ 11.94
7 - R$ 13.93
8 - R$ 15.92
9 - R$ 17.91
10 - R$ 19.9
11 - R$ 21.89
12 - R$ 23.88
13 - R$ 25.87
14 - R$ 27.86
15 - R$ 29.85
16 - R$ 31.84
17 - R$ 33.83
18 - R$ 35.82
19 - R$ 37.81
20 - R$ 39.8
21 - R$ 41.79
22 - R$ 43.78
23 - R$ 45.77
24 - R$ 47.76
25 - R$ 49.75
26 - R$ 51.74
27 - R$ 53.73
28 - R$ 55.72
29 - R$ 57.71
30 - R$ 59.7
31 - R$ 61.69
32 - R$ 63.68
33 - R$ 65.67
34 - R$ 67.66
35 - R$ 69.65
36 - R$ 71.64
37 - R$ 73.63
38 - R$ 75.62
39 - R$ 77.61
40 - R$ 79.6
41 - R$ 81.59
42 - R$ 83.58
43 - R$ 85.57
44 - R$ 87.56
45 - R$ 89.55
46 - R$ 91.54
47 - R$ 93.53
48 - R$ 95.52
49 - R$ 97.51
50 - R$ 99.5


21. O Sr. Manoel Joaquim acaba de adquirir uma panificadora e pretende implantar a metodologia da tabelinha, que já é um sucesso na sua loja de 1,99. Você foi contratado para desenvolver o programa que monta a tabela de preços de pães, de
1 até 50 pães, a partir do preço do pão informado pelo usuário, conforme o exemplo abaixo:
Preço do pão: R$ 0.18
Panificadora Pão de Ontem - Tabela de preços
1 - R$ 0.18
2 - R$ 0.36
...
50 - R$ 9.00


In [37]:
pao = float(input('Digite o valor do pão:'))
i =1
while i <= 50:
    preco = i * pao
    print(" %.0f - R$ %.2f" % (i, preco))
    i += 1

Digite o valor do pão:0.35
 1 - R$ 0.35
 2 - R$ 0.70
 3 - R$ 1.05
 4 - R$ 1.40
 5 - R$ 1.75
 6 - R$ 2.10
 7 - R$ 2.45
 8 - R$ 2.80
 9 - R$ 3.15
 10 - R$ 3.50
 11 - R$ 3.85
 12 - R$ 4.20
 13 - R$ 4.55
 14 - R$ 4.90
 15 - R$ 5.25
 16 - R$ 5.60
 17 - R$ 5.95
 18 - R$ 6.30
 19 - R$ 6.65
 20 - R$ 7.00
 21 - R$ 7.35
 22 - R$ 7.70
 23 - R$ 8.05
 24 - R$ 8.40
 25 - R$ 8.75
 26 - R$ 9.10
 27 - R$ 9.45
 28 - R$ 9.80
 29 - R$ 10.15
 30 - R$ 10.50
 31 - R$ 10.85
 32 - R$ 11.20
 33 - R$ 11.55
 34 - R$ 11.90
 35 - R$ 12.25
 36 - R$ 12.60
 37 - R$ 12.95
 38 - R$ 13.30
 39 - R$ 13.65
 40 - R$ 14.00
 41 - R$ 14.35
 42 - R$ 14.70
 43 - R$ 15.05
 44 - R$ 15.40
 45 - R$ 15.75
 46 - R$ 16.10
 47 - R$ 16.45
 48 - R$ 16.80
 49 - R$ 17.15
 50 - R$ 17.50


22. O Sr. Manoel Joaquim expandiu seus negócios para além dos negócios de 1,99 e agora possui uma loja de conveniências. Faça um programa que implemente uma caixa registradora rudimentar. O programa deverá receber um número desconhecido de valores referentes aos preços das mercadorias. Um valor zero deve ser informado pelo operador para indicar o final da compra. O programa deve então mostrar o total da compra e perguntar o valor em dinheiro que o cliente forneceu, para então calcular e mostrar o valor do troco. Após esta operação, o programa deverá voltar ao ponto inicial, para registrar a próxima compra. A saída deve ser conforme o exemplo abaixo:
Lojas Tabajara
Produto 1: R$ 2.20
Produto 2: R$ 5.80
Produto 3: R$ 0
Total: R$ 9.00
Dinheiro: R$ 20.00
Troco: R$ 11.00

In [1]:
total = 0
preco = -1
i = 0
while preco != 0:
    preco = float(input("Insira o preço do item (Digite '0' para finalizar a compra): "))
    total += preco
    i += 1
    if preco == 0:
        print(f"Total: R$ {total}")
        pago = float(input("Insira o valor a ser pago: "))
        print("Troco: R$ ", pago - total)
    else:
        print(f"Produto {i}: R$ {preco}")

Insira o preço do item (Digite '0' para finalizar a compra): 8
Produto 1: R$ 8.0
Insira o preço do item (Digite '0' para finalizar a compra): 4
Produto 2: R$ 4.0
Insira o preço do item (Digite '0' para finalizar a compra): 2
Produto 3: R$ 2.0
Insira o preço do item (Digite '0' para finalizar a compra): 0
Total: R$ 14.0
Insira o valor a ser pago: 20
Troco: R$  6.0


23. Foi feita uma estatística em cinco cidades brasileiras para coletar dados sobre acidentes de trânsito. Foram obtidos os seguintes dados:
- Código da cidade;
- Número de veículos de passeio (em 1999);
- Número de acidentes de trânsito com vítimas (em 1999). Deseja-se saber:
- Qual o maior e menor índice de acidentes de transito e a que cidade pertence;
- Qual a média de veículos nas cinco cidades juntas;
- Qual a média de acidentes de trânsito nas cidades com menos de 2.000 veículos depasseio.

In [4]:
cidades = {
    "cidade0": [6753, 12],
    "cidade1": [7503, 23],
    "cidade2": [607, 3],
    "cidade3": [450, 2],
    "cidade4": [12045, 47]
}

menorIndice = [cidades["cidade0"][0], cidades["cidade0"][1]]
maiorIndice = [cidades["cidade0"][0], cidades["cidade0"][1]]
somaVeiculos = 0
somaAcidentes = 0
contCidades = 0
for cod in cidades:
    somaVeiculos += cidades[cod][0]
    if cidades[cod][0] < 2000:
        somaAcidentes += cidades[cod][1]
        contCidades += 1
    if cidades[cod][1] > maiorIndice[0]:
        maiorIndice[0] = cidades[cod][1]
        maiorIndice[1] = cod
    if cidades[cod][1] < menorIndice[0]:
        menorIndice[0] = cidades[cod][1]
        menorIndice[1] = cod
        
    
mediaVeiculos = somaVeiculos / len(cidades.keys())
mediaAcidentes = somaAcidentes / contCidades

print("Maior índice e a cidade: ", maiorIndice)
print("Menor índice e a cidade: ", menorIndice)
print("Média de veículos nas cinco cidades juntas: ", mediaVeiculos)
print("Média de acidentes de trânsito nas cidades com menos de 2.000 veículos de passeio: ", mediaAcidentes)

Maior índice e a cidade:  [6753, 12]
Menor índice e a cidade:  [2, 'cidade3']
Média de veículos nas cinco cidades juntas:  5471.6
Média de acidentes de trânsito nas cidades com menos de 2.000 veículos de passeio:  2.5


24. Faça um programa que receba o valor de uma dívida e mostre uma tabela com os seguintes dados: valor da dívida, valor dos juros, quantidade de parcelas e valor da parcela. Os juros e a quantidade de parcelas seguem a tabela abaixo:
Quantidade de Parcelas % de Juros sobre o valor inicial da dívida
1 0
3 10
6 15
9 20
12 25
Exemplo de saída do programa:
Valor da Dívida Valor dos Juros Quantidade de Parcelas Valor da Parcela
R$ 1.000,00 0 1 R$ 1.000,00
R$ 1.100,00 100 3 R$ 366,00
R$ 1.150,00 150 6 R$ 191,67

In [5]:
dividaInicial = float(input("Insira o valor da dívida: "))
print("Dívida \t Valor Juros \t Qtde Parcelas \t Valor da Parcela")
for i in range(0, 5):
    if i == 0:
        juros = 0
    elif i == 1:
        juros = 10
    else:
        juros += 5
    valorJuros = dividaInicial * (juros / 100)
    divida = dividaInicial + valorJuros
    if i == 0:
        qtdeParc = 1
    else:
        qtdeParc = i * 3
    if qtdeParc != 0:
        valorParcela = divida / qtdeParc
    else:
        valorParcela = divida
    print(f"R$ {divida} \t {valorJuros} \t {qtdeParc} \t R$ %.2f"%(valorParcela))

Insira o valor da dívida: 2500
Dívida 	 Valor Juros 	 Qtde Parcelas 	 Valor da Parcela
R$ 2500.0 	 0.0 	 1 	 R$ 2500.00
R$ 2750.0 	 250.0 	 3 	 R$ 916.67
R$ 2875.0 	 375.0 	 6 	 R$ 479.17
R$ 3000.0 	 500.0 	 9 	 R$ 333.33
R$ 3125.0 	 625.0 	 12 	 R$ 260.42


25. Faça um programa que mostre os n termos da Série a seguir:
 S = 1/1 + 2/3 + 3/5 + 4/7 + 5/9 + ... + n/m.
Imprima no final a soma da série.

In [8]:
termo = int(input('Insira a quantidade de termos desejada: '))

soma = "S = "
s = 0
m = 1
for n in range(1, termo + 1):
    s += n / m
    soma += str(n) + "/" + str(m)
    if n != termo:
        soma += " + "
    m += 2

print(somaStr)
print('Soma da série: %.2f' % s)

Insira a quantidade de termos desejada: 5
S = 1/1 + 2/3 + 3/5 + 4/7 + 5/9
Soma da série: 3.39
